torch/_inductor/pattern_matcher.py

```diff
@@ -919,15 +929,53 @@ def check_fn(match: Match):
                         device=args[i].device,
                         requires_grad=grad,
                     )
-            try:
-                specific_graph = trace_fn(search_fn, args)
-            except RuntimeError as e:
-                log.info(
-                    "Replacement pattern %s failed to apply due to shape mismatch: %s",
-                    search_fn.__name__,
-                    e,
-                )
-                return False
+                    for v in itertools.chain(args[i].shape, args[i].stride()):
+                        if isinstance(v, torch.SymInt) and v not in sym_args:
```

Comment:
This is highly suspicious. Python will do an equality test when testing for membership, so `v not in sym_args` will effectively install a `!=` guard on every entry in `sym_args`. I can't easily tell what this logic is trying to do in reality though...

Reply:
It's gathering all of the unique symints as inputs. as stated below:

```
                # AOT Autograd and make fx will dedupe symbolic shape size
                # accesses of sym ints that appear as inputs
                # We don't want the sym_size uses to interfere with pattern matching
                # so we provide them as inputs.
 ```